internal/provider/provider.go

```diff
@@ -87,13 +85,6 @@ func AzureADProvider() *schema.Provider {
 				Description: "The Tenant ID which should be used. Works with all authentication methods except Managed Identity.",
 			},
 
-			"metadata_host": {
```

Comment:
At the moment this isn't supported, since the metadata service doesn't publish MS Graph URIs so we no longer consult it. I'm not even sure if MS Graph exists outside of the published environments, but if it does, I was thinking we could add a provider field later to specify a custom URI as a workaround?

Reply:
👍 sounds good - you may want to reach out to our contact wrt air gapped envs at microsoft to double check before we release however?

Reply:
Will do, I'll open another PR to add custom support before releasing